### PR TITLE
Add generic unsupported elements handling steps and implement the fist use for the Salesforce connector

### PR DESCRIPTION
**Pull request by jesica-fera**
* Also fix unsupported namespace handling to comment unsupported elements instead of leaving them with just an error

On the salesforce connector this is fixing cases using config-oauth-user-pass for now because it was reported but more elements can be added if needed

### CODE REVIEW

**Comment by jesica-fera** [comment]
This is a fix for https://github.com/mulesoft/mule-migration-assistant/issues/564

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/salesforce_config_oauth_user_pass/output/report/report.json [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  "numberOfMELExpressionLinesMigrated": 0,
+  "numberOfDWTransformations": 0,
+  "numberOfDWTransformationsMigrated": 0,
+  "numberOfDWTransformationLines": 0,
+  "numberOfDWTransformationLinesMigrated": 0,
+  "detailedMessages": [
+    {
+      "level": "ERROR",
+      "key": "components.unsupported",
+      "component": "mule",
```
Comment (svacas) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/salesforce_config_oauth_user_pass/output/report/report.json:31]: should report `sfdc:config-oauth-user-pass` as the error component with its line and col numbers

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/salesforce_config_oauth_user_pass/output/report/report.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      "lineNumber": 2,
+      "columnNumber": 292,
+      "message": "The migration of \u0027sfdc:config-oauth-user-pass (name = Salesforce__Basic_Authentication)\u0027 is not supported.",
+      "filePath": "src/main/mule/mule-config.xml",
+      "documentationLinks": []
+    },
+    {
+      "level": "ERROR",
+      "key": "components.unsupported",
+      "component": "mule",
```
Comment (svacas) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/salesforce_config_oauth_user_pass/output/report/report.json:41]: should report `sfdc:config-oauth-user-pass` as the error component with its line and col numbers

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/unknown_namespace/output/report/report.json [hunk 3]
```diff
@@ -38,22 +38,24 @@
     {
       "level": "ERROR",
       "key": "components.unsupported",
-      "component": "unknown:source",
-      "lineNumber": 5,
-      "columnNumber": 25,
-      "message": "The migration of \u0027unknown\u0027 is not supported.",
+      "component": "flow",
```
Comment (svacas) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/unknown_namespace/output/report/report.json:41]: should report `unknown:source` as the error component with its line and col numbers

File: mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/unknown_namespace/output/report/report.json [hunk 4]
```diff
@@ -38,22 +38,24 @@
...
+      []
     },
     {
       "level": "ERROR",
       "key": "components.unsupported",
-      "component": "unknown:operation",
-      "lineNumber": 16,
-      "columnNumber": 28,
-      "message": "The migration of \u0027unknown\u0027 is not supported.",
+      "component": "flow",
```
Comment (svacas) [inline mule-migration-tool-e2e-tests/src/test/resources/e2e/gaps/unknown_namespace/output/report/report.json:52]: should report `unknown:operation` as the error component with its line and col numbers

**Review by svacas — CHANGES_REQUESTED** [review]

**Comment by jesica-fera** [comment]
I am closing this PR in favor of this one https://github.com/mulesoft/mule-migration-assistant/pull/614 because fixing this properly requires a bigger refactor. When the components are commented out, reporting the right element and location becomes complex because there are several cases where elements are mutated so location and parent element are not easy to calculate. For now I am creating the second PR to fix the actual original issue.

**Pull request closed without merging**